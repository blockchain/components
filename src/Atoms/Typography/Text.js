import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { keysIn } from 'ramda'
import theme from '../../Tools/theme'

const BaseText = styled.span`
  font-family: 'Montserrat', helvetica, sans serif;
  font-weight: ${(props) => props.weight};
  font-size: ${(props) => props.size};
  text-transform: ${(props) =>
    props.uppercase ? 'uppercase' : props.capitalize ? 'capitalize' : 'none'};
  font-style: ${(props) => (props.italic ? 'italic' : 'normal')};
  color: ${(props) => props.theme[props.color]};
  opacity: ${(props) => props.opacity};
`

const Text = ({
  children,
  weight,
  size,
  uppercase,
  capitalize,
  italic,
  color,
  opacity,
  ...rest
}) => (
  <BaseText
    capitalize={capitalize}
    color={color}
    italic={italic}
    opacity={opacity}
    size={size}
    uppercase={uppercase}
    weight={weight}
    {...rest}
  >
    {children}
  </BaseText>
)

Text.propTypes = {
  capitalize: PropTypes.bool,
  children: PropTypes.node,
  color: PropTypes.oneOf(keysIn(theme)),
  italic: PropTypes.bool,
  opacity: PropTypes.number,
  size: PropTypes.string,
  uppercase: PropTypes.bool,
  weight: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700, 800, 900]),
}

Text.defaultProps = {
  capitalize: false,
  color: 'gray-5',
  italic: false,
  opacity: 1,
  size: '14px',
  uppercase: false,
  weight: 300,
}

export default Text
