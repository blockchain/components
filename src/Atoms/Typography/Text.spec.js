import React from 'react'
import { mount } from 'enzyme'

import Text from './Text'

describe('Text', () => {
  it('should render correctly', () => {
    const tree = mount(<Text>MyText</Text>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with color', () => {
    const tree = mount(<Text color="gold">MyText</Text>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with size', () => {
    const tree = mount(<Text size="32px">MyText</Text>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with weight', () => {
    const tree = mount(<Text weight={500}>MyText</Text>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with uppercase', () => {
    const tree = mount(<Text uppercase>MyText</Text>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with capitalize', () => {
    const tree = mount(<Text capitalize>MyText</Text>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with italic', () => {
    const tree = mount(<Text italic>MyText</Text>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with opacity', () => {
    const tree = mount(<Text opacity={0.5}>MyText</Text>)

    expect(tree).toMatchSnapshot()
  })
})
