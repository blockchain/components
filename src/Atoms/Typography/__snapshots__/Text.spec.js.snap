// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render correctly 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Text
    capitalize={false}
    italic={false}
    opacity={1}
    size="1rem"
    uppercase={false}
    weight={400}
  >
    <styled.span
      capitalize={false}
      italic={false}
      opacity={1}
      size="1rem"
      uppercase={false}
      weight={400}
    >
      <span
        className="c0"
        opacity={1}
        size="1rem"
      >
        MyText
      </span>
    </styled.span>
  </Text>
</ThemeProvider>
`;

exports[`Text should render correctly with capitalize 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: capitalize;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Text
    capitalize={true}
    italic={false}
    opacity={1}
    size="1rem"
    uppercase={false}
    weight={400}
  >
    <styled.span
      capitalize={true}
      italic={false}
      opacity={1}
      size="1rem"
      uppercase={false}
      weight={400}
    >
      <span
        className="c0"
        opacity={1}
        size="1rem"
      >
        MyText
      </span>
    </styled.span>
  </Text>
</ThemeProvider>
`;

exports[`Text should render correctly with italic 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: none;
  font-style: italic;
  color: #545456;
  opacity: 1;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Text
    capitalize={false}
    italic={true}
    opacity={1}
    size="1rem"
    uppercase={false}
    weight={400}
  >
    <styled.span
      capitalize={false}
      italic={true}
      opacity={1}
      size="1rem"
      uppercase={false}
      weight={400}
    >
      <span
        className="c0"
        opacity={1}
        size="1rem"
      >
        MyText
      </span>
    </styled.span>
  </Text>
</ThemeProvider>
`;

exports[`Text should render correctly with opacity 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 0.5;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Text
    capitalize={false}
    italic={false}
    opacity={0.5}
    size="1rem"
    uppercase={false}
    weight={400}
  >
    <styled.span
      capitalize={false}
      italic={false}
      opacity={0.5}
      size="1rem"
      uppercase={false}
      weight={400}
    >
      <span
        className="c0"
        opacity={0.5}
        size="1rem"
      >
        MyText
      </span>
    </styled.span>
  </Text>
</ThemeProvider>
`;

exports[`Text should render correctly with size 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 32px;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Text
    capitalize={false}
    italic={false}
    opacity={1}
    size="32px"
    uppercase={false}
    weight={400}
  >
    <styled.span
      capitalize={false}
      italic={false}
      opacity={1}
      size="32px"
      uppercase={false}
      weight={400}
    >
      <span
        className="c0"
        opacity={1}
        size="32px"
      >
        MyText
      </span>
    </styled.span>
  </Text>
</ThemeProvider>
`;

exports[`Text should render correctly with uppercase 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: uppercase;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Text
    capitalize={false}
    italic={false}
    opacity={1}
    size="1rem"
    uppercase={true}
    weight={400}
  >
    <styled.span
      capitalize={false}
      italic={false}
      opacity={1}
      size="1rem"
      uppercase={true}
      weight={400}
    >
      <span
        className="c0"
        opacity={1}
        size="1rem"
      >
        MyText
      </span>
    </styled.span>
  </Text>
</ThemeProvider>
`;

exports[`Text should render correctly with weight 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 500;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#383838",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <Text
    capitalize={false}
    italic={false}
    opacity={1}
    size="1rem"
    uppercase={false}
    weight={500}
  >
    <styled.span
      capitalize={false}
      italic={false}
      opacity={1}
      size="1rem"
      uppercase={false}
      weight={500}
    >
      <span
        className="c0"
        opacity={1}
        size="1rem"
      >
        MyText
      </span>
    </styled.span>
  </Text>
</ThemeProvider>
`;
